TST: apply_space_motion no longer emits ErfaWarning in remote data/dev job

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/master/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/master/CODE_OF_CONDUCT.md . -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- Please check that the development version still produces the same bug.
You can install development version with
pip install git+https://github.com/astropy/astropy
command. -->

Example log: https://travis-ci.org/github/astropy/astropy/jobs/717518434 (first 2 failures)

```
____________________________ test_regression_10092 _____________________________
    def test_regression_10092():
        """
        Check that we still get a proper motion even for SkyCoords without distance
        """
        c = SkyCoord(l=10*u.degree, b=45*u.degree,
                     pm_l_cosb=34*u.mas/u.yr, pm_b=-117*u.mas/u.yr,
                     frame='galactic',
                     obstime=Time('1988-12-18 05:11:23.5'))
    
        with pytest.warns(ErfaWarning, match='ERFA function "pmsafe" yielded .*'):
            # expect ErfaWarning here
>           newc = c.apply_space_motion(dt=10*u.year)
E           Failed: DID NOT WARN. No warnings of type (<class 'erfa.core.ErfaWarning'>,) matching ('ERFA function "pmsafe" yielded .*') was emitted. The list of emitted warnings is: [ErfaWarning('ERFA function \'pmsafe\' yielded 1 of "distance overridden (Note 6)"')].
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/coordinates/tests/test_regression.py:641: Failed
___________________________ test_apply_space_motion ____________________________
    def test_apply_space_motion():
        # use this 12 year period because it's a multiple of 4 to avoid the quirks
        # of leap years while having 2 leap seconds in it
        t1 = Time('2000-01-01T00:00')
        t2 = Time('2012-01-01T00:00')
    
        # Check a very simple case first:
        frame = ICRS(ra=10.*u.deg, dec=0*u.deg,
                     distance=10.*u.pc,
                     pm_ra_cosdec=0.1*u.deg/u.yr,
                     pm_dec=0*u.mas/u.yr,
                     radial_velocity=0*u.km/u.s)
    
        # Cases that should work (just testing input for now):
        c1 = SkyCoord(frame, obstime=t1, pressure=101*u.kPa)
        with pytest.warns(ErfaWarning, match='ERFA function "pmsafe" yielded .*'):
            # warning raised due to high PM chosen above
            applied1 = c1.apply_space_motion(new_obstime=t2)
>           applied2 = c1.apply_space_motion(dt=12*u.year)
E           Failed: DID NOT WARN. No warnings of type (<class 'erfa.core.ErfaWarning'>,) matching ('ERFA function "pmsafe" yielded .*') was emitted. The list of emitted warnings is: [ErfaWarning('ERFA function \'pmsafe\' yielded 1 of "distance overridden (Note 6)"'), ErfaWarning('ERFA function \'pmsafe\' yielded 1 of "distance overridden (Note 6)"')].
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/coordinates/tests/test_sky_coord.py:1613: Failed
```

If the warning is no longer emitted on purpose, `pytest.warns` needs to be removed. Otherwise, if a warning is expected, this is a symptom of a bug.

--- a/astropy/coordinates/tests/test_regression.py
+++ b/astropy/coordinates/tests/test_regression.py
@@ -10,6 +10,7 @@ import io
 import copy
 import pytest
 import numpy as np
+from erfa import ErfaWarning
 
 from astropy import units as u
 from astropy.coordinates import (
@@ -22,7 +23,6 @@ from astropy.coordinates import (
     CylindricalRepresentation, CylindricalDifferential,
     CartesianDifferential)
 from astropy.coordinates.sites import get_builtin_sites
-from astropy.utils.exceptions import ErfaWarning
 from astropy.time import Time
 from astropy.utils import iers
 from astropy.table import Table

--- a/astropy/coordinates/tests/test_sky_coord.py
+++ b/astropy/coordinates/tests/test_sky_coord.py
@@ -12,6 +12,7 @@ from copy import deepcopy
 import pytest
 import numpy as np
 import numpy.testing as npt
+from erfa import ErfaWarning
 
 from astropy import units as u
 from astropy.tests.helper import assert_quantity_allclose as assert_allclose
@@ -25,7 +26,6 @@ from astropy.coordinates import Latitude, EarthLocation
 from astropy.coordinates.transformations import FunctionTransform
 from astropy.time import Time
 from astropy.utils import minversion, isiterable
-from astropy.utils.exceptions import ErfaWarning
 from astropy.units import allclose as quantity_allclose
 from astropy.io import fits
 from astropy.wcs import WCS

--- a/astropy/time/tests/test_basic.py
+++ b/astropy/time/tests/test_basic.py
@@ -11,8 +11,9 @@ import numpy as np
 import pytest
 from numpy.testing import assert_allclose
 import erfa
+from erfa import ErfaWarning
 
-from astropy.utils.exceptions import AstropyDeprecationWarning, ErfaWarning
+from astropy.utils.exceptions import AstropyDeprecationWarning
 from astropy.utils import isiterable, iers
 from astropy.time import (Time, TimeDelta, ScaleValueError, STANDARD_TIME_SCALES,
                           TimeString, TimezoneInfo, TIME_FORMATS)

--- a/astropy/time/tests/test_precision.py
+++ b/astropy/time/tests/test_precision.py
@@ -13,13 +13,13 @@ from hypothesis.strategies import (composite, datetimes, floats, integers,
 
 import numpy as np
 import erfa
+from erfa import ErfaError, ErfaWarning
 
 import astropy.units as u
 from astropy.tests.helper import assert_quantity_allclose
 from astropy.time import STANDARD_TIME_SCALES, Time, TimeDelta
 from astropy.time.utils import day_frac, two_sum
 from astropy.utils import iers
-from astropy.utils.exceptions import ErfaError, ErfaWarning
 
 
 allclose_jd = functools.partial(np.allclose, rtol=np.finfo(float).eps, atol=0)

--- a/astropy/visualization/tests/test_time.py
+++ b/astropy/visualization/tests/test_time.py
@@ -6,10 +6,10 @@ import pytest
 pytest.importorskip('matplotlib')  # noqa
 
 import matplotlib.pyplot as plt
+from erfa import ErfaWarning
 
 from astropy.time import Time
 from astropy.visualization.time import time_support
-from astropy.utils.exceptions import ErfaWarning
 from astropy.utils.compat.context import nullcontext
 
 # Since some of the examples below use times/dates in the future, we use the

